Terraform base script

gopal-kannan opened the pull request:
Working base script creating necessary components.

gopal-kannan committed:
tf file upload

test for calling from teamcity

gopal-kannan committed:
TF files for basic infrastructure items

gopal-kannan committed:
Delete azure_stg_acc.tf

gopal-kannan committed:
Update main.tf

gopal-kannan committed:
Update main.tf

gopal-kannan committed:
Update main.tf

gopal-kannan committed:
Tested working version

Terraform scripts are tested thru Azure bash.

gopal-kannan committed:
Commented the databricks github configuration

s-laugh commented on README.md:
```diff
@@ -1,10 +1 @@
-- [AP-Databricks](https://github.com/DTS-STN/AP-Databricks)
-- [AP-DataFactory](https://github.com/DTS-STN/AP-DataFactory)
-- AP-Synapse (to be created)
+# saeb
```
This file shouldn't get overridden like this.

gopal-kannan replied:
Necessary content is added.

s-laugh replied:
I'm not really clear on the need for the changes in this file - it's now adding a list of files, but that can be found though the directory and this PR/commit history, the readme shouldn't need to identify them; and their is already a header.

gopal-kannan replied:
This document should be living document and grows as we make changes. If more changes required, we can discuss it and determine what are all the content to go and instructions to the developers what to add as and when they commit new changes.

gopal-kannan replied:
I added some content. Please let me know your feedback.

s-laugh replied:
looks good

s-laugh replied:
looks good

s-laugh commented on main.tf:
```diff
@@ -0,0 +1,116 @@
+locals {
+  build_environment = var.environment == "production" ? "Prod" : "Dev"
+  common_tags = {
+    Product      = "SAEB Platform"
```
```suggestion
    ProductPortfolio = "SAEB"
    Product      = "SAEB Analytics Platform"
```

s-laugh commented on main.tf:
```diff
@@ -0,0 +1,116 @@
+
+# Create storage account for main input and curated CSV files as well as for synapse filesystem
+resource "azurerm_storage_account" "saeb_storage" {
+  name                     = "stsaebdevca01${random_string.random.result}"
```
We might have to loop back on our naming conventions unless we can include a dynamic environment component to the name.
```suggestion
  name                     = "stapsndbxca01${random_string.random.result}"
```

s-laugh reviewed (changes requested):
Looks good! Just some minor tweaks.
We can loop back in another place about managing the naming conventions and ensuring the env part is injected accurately.

gopal-kannan committed:
Update main.tf

Co-authored-by: Shaun Laughland <22419216+s-laugh@users.noreply.github.com>

gopal-kannan committed:
Update main.tf

Co-authored-by: Shaun Laughland <22419216+s-laugh@users.noreply.github.com>

gopal-kannan committed:
Initial Terraform version

gopal-kannan commented:
Please review and let me know your comments if any.

s-laugh reviewed (commented):
I'm not understanding the changes to the readme; also wanted to double check that all the names for resources were reviewed to match the latest in the naming conventions (i only saw the one change I suggested).

gopal-kannan committed:
Updated the file with formatted content

gopal-kannan committed:
Update README.md

gopal-kannan committed:
Updated with common content 

This document should be living document and grows as we make changes. If more changes required, we can discuss it and determine what are all the content to go and instructions to the developers what to add as and when they commit new changes.

s-laugh commented on main.tf:
```diff
@@ -0,0 +1,117 @@
+# Declare some local variables
+locals {
+  build_environment = var.environment == "production" ? "Prod" : "Dev"
```
glad we have a variable for this, but I don't see it being used anywhere and it's missing a "Sndbx" state - it should effect the name of each resource and be a common tag, the tag should have the full name "Sandbox", "Development", "Production".

gopal-kannan replied:
The environment variable configuration is complete. The github configuration for development is still pending due to the challenge we have with firewall / proxy.

gopal-kannan committed:
Update main.tf

gopal-kannan committed:
Environment variable check and github config

gopal-kannan committed:
Update terraform.tfvars

gopal-kannan committed:
Update main.tf

gopal-kannan commented:
This is the most recent version of tf code which can go to main branch. @s-laugh please approve the merge so we have a base version saved in main branch. Thanks.

gopal-kannan commented:
@s-laugh  I am not authorized to merge this pull request to main branch. Could you please merge it? Thanks.
